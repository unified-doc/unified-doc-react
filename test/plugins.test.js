import React from 'react';
import renderer from 'react-test-renderer';
import toc from 'rehype-toc';

import { htmlContent } from './fixtures';
import Doc from '../lib/component';

describe('plugins', () => {
  it('applies custom plugins', () => {
    const plugins = [[toc, { cssClasses: { list: 'custom-list' } }]];
    const sanitizeSchema = { attributes: { '*': ['className'] } };
    const tree = renderer
      .create(
        <Doc
          content={htmlContent}
          filename="doc.html"
          plugins={plugins}
          sanitizeSchema={sanitizeSchema}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
