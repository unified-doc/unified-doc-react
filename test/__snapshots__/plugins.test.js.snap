// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins applies custom plugins 1`] = `
<div>
  <div>
    <ol
      className="custom-list custom-list-1"
    />
    <blockquote>
      <strong>
        some
      </strong>
      
content
    </blockquote>
  </div>
</div>
`;
